scrapy/extensions/robotstxtparser.py

```diff
@@ -0,0 +1,52 @@
+from six.moves.urllib_robotparser import RobotFileParser
+
+from scrapy.utils.python import to_native_str
+
+class BaseRobotsTxtParser():
+    def __init__(self, robotstxt_url):
+        raise NotImplementedError
+
+    def parse(self, robotstxt_body):
+        """Parse content of robots.txt file."""
+        raise NotImplementedError
+
+    def allowed(self, url, useragent):
+        """Return True if url is allowed for crawling, otherwise return False."""
+        raise NotImplementedError
+
+    def sitemaps(self):
+        """Return a list of links of sitemaps on the website. If there is no sitemap specified in robots.txt, return None."""
+        raise NotImplementedError
+
+    def crawl_delay(self, useragent):
+        """Return time specified with Crawl-delay directive. If nothing is specified, return None."""
```

Comment:
The documentation should probably clarify the unit of the result (e.g. seconds).